### PR TITLE
Routes and endpoints

### PR DESCRIPTION
**Pull request by mikez321**
### Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling (Styling to new or existing files)
- [ ] Testing
### Detailed Description
Create an endpoint for all plants, plant by ID, and most importantly, MEET A PLANT!
### Why is this change required? What problem does it solve?
This allows us to pass a real endpoint to our FE team!
### Was any previously written code changed?
We added the above mentioned endpoint

### CODE REVIEW

File: alembic/versions/752c338c6328_seed_db.py [hunk 1]
```diff
@@ -33,30 +33,30 @@ def upgrade():
...
+        {'id':17,'name':'Todd','plant_type':'Lemon','image':'https://p0.pikrepo.com/preview/784/509/two-lemons-and-1-slice-of-lemon.jpg','lighting':'Full Sun','water_frequency':4,'harvest_time':365,'root_depth':24,'annual':False},
+        {'id':18,'name':'Leslie','plant_type':'Lime','image':'https://i2.pickpik.com/photos/197/226/505/sliced-lime-fruit-lime-slice-preview.jpg','lighting':'Full Sun','water_frequency':4,'harvest_time':121,'root_depth':24,'annual':False},
+        {'id':19,'name':'Henri','plant_type':'Grapes','image':'https://p0.pikist.com/photos/670/14/autumn-grapes-grape-vine-fruit-winegrowing-grapevine-blue-grapes-wine-harvest.jpg','lighting':'Full Sun','water_frequency':7,'harvest_time':1095,'root_depth':36,'annual':False},
+        {'id':20,'name':'Jennifer','plant_type':'Asparagus','image':'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcS2HpJaaQQPpaDrolZY-cg9owfb4z9oy2eZ-A&usqp=CAU','lighting':'Full Sun','water_frequency':21,'harvest_time':730,'root_depth':10,'annual':False},
```
Comment (kelshadarby) [inline alembic/versions/752c338c6328_seed_db.py:55]: Nice catch here with the root_depth!

File: app/__init__.py [hunk 2]
```diff
@@ -18,8 +18,7 @@ def create_app(config_name):
     app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
     db.init_app(app)
 
-    @app.route('/', methods=['GET'])
-    def endpoint():
-        return "I need to go take a shower so I can't tell if I'm crying or not."
+    from .api import api as api_blueprint
+    app.register_blueprint(api_blueprint, url_prefix='/api/v1')
```
Comment (kelshadarby) [inline app/__init__.py:22]: Finally got the routes out of the app.py 🥳

File: app/api/routes.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+from app import db
+
+
+@api.route('/')
+def endpoint():
+    return "I need to go take a shower so I can't tell if I'm crying or not."
+
+@api.route('/plants/<int:id>')
+def get_plant(id):
+    plant = Plant.query.filter_by(id=id).first()
```
Comment (kelshadarby) [inline app/api/routes.py:14]: We might want to get this into a model when refacotring

File: alembic/versions/752c338c6328_seed_db.py (hunk 1 above)
Comment (mikez321) [reply alembic/versions/752c338c6328_seed_db.py:55]: I'm actually surprised it let us migrate that... Something to go back and fix later!

File: app/api/routes.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        }
+        results.append(obj)
+    response = jsonify(results)
+    response.status_code = 200
+    return response
+
+@api.route('/plants/meet')
+def random_plant():
+    from random import randint
+    id = randint(1, 24)
```
Comment (kelshadarby) [inline app/api/routes.py:42]: We should find a way to get the number of records in the table to generate a random number off of. I think we could do this with a get_all().count or something similar

File: app/api/routes.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                'plant_type': plant.plant_type,
+                'plant_image': plant.image,
+                'lighting': plant.lighting,
+                'days_between_water': plant.water_frequency,
+                'days_to_harvest_from_seed': plant.harvest_time,
+                'root_depth_in': plant.root_depth,
+                'annual?': plant.annual
+                }
+    response = jsonify(result)
+    response.status_code = 200
```
Comment (kelshadarby) [inline app/api/routes.py:56]: We can also do this using just the result object and avoid the jsonify method editing the order of the result

File: app/models.py [hunk 6]
```diff
@@ -11,3 +12,9 @@ class Plant(db.Model):
     harvest_time = db.Column(db.Integer, nullable=True)
     root_depth = db.Column(db.Integer)
     annual = db.Column(db.Boolean)
+
+    def get_all():
+        return Plant.query.all()
+
+    def get_one(id):
+        return Plant.query.filter_by(id=id).first()
```
Comment (kelshadarby) [inline app/models.py:20]: Nice! Maybe we can combine randint and db calls into a model method

File: tests/test_endpoint.py [hunk 7]
```diff
@@ -22,10 +22,18 @@ def tearDown(self):
...
     def test_api_can_get_a_dummy_endpoint(self):
         """Test API can get a dummy string (GET request)."""
-        res = self.client().get('/')
+        res = self.client().get('/api/v1/')
         self.assertEqual(res.status_code, 200)
         self.assertIn("I need to go take a shower so I can't tell if I'm crying or not.", str(res.data))
 
+    def test_api_can_return_a_plant_by_id(self):
+        jimmy = Plant(name='Jimothy',plant_type='Cherry Tomato',image='jim_photo.jpg',lighting='Full Sun',water_frequency=3,harvest_time=50,root_depth=12,annual=False)
+        dan = Plant(name='Dan',plant_type='Cactus',image='cactus_dan.jpg',lighting='Full Sun',water_frequency=7,harvest_time=None,root_depth=8,annual=False)
```
Comment (kelshadarby) [inline tests/test_endpoint.py:31]: LOVE cactus Dan, what a guy

File: app/api/routes.py (hunk 4 above)
Comment (mikez321) [reply app/api/routes.py:42]: LOVE IT.

**Pull request merged**